flowey/flowey_lib_hvlite/src/init_cross_arch_build.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Install dependencies and set environment variables necessary for cross
+//! compiling to another architecture.
+
+use flowey::node::prelude::*;
+use std::collections::BTreeMap;
+
+flowey_request! {
+    pub struct Request{
+        pub target: target_lexicon::Triple,
+        pub cross_deps_write: WriteVar<SideEffect>,
+        pub injected_env_write: Option<WriteVar<BTreeMap<String, String>>>,
```

Comment:
I would model these two fields as an enum with two variants. either you want the side-effect, or you want the env-vars. doing so should simplify downstream code

Reply:
alternatively, you can just unconditionally return `injected_env`, and have callers use `into_side_effect()`.

yeah, maybe just do that. seems easier?

Reply:
I guess there's no reason I couldn't split it up into two separate requests. I'll give that a try and if I get too complicated I'll just do the into_side_effect() thing.

Reply:
I refactored it into separate requests. I like how it turned out. I also made it platform-general, so it is clear what is supported.